.github/workflows/build.yml

```diff
@@ -0,0 +1,95 @@
+name: Validate Build
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  docker-validate:
+    name: Validate Build
+    runs-on: ubuntu-latest
+    services:
+      registry:
+        image: docker.io/library/registry
+        ports:
+          - 5000:5000
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.24
+      - uses: arduino/setup-protoc@v3
+        with:
+          version: 31.1
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Install Tools
+        run: |
+          go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.36
```

Comment:
Out of curiosity, why are these tools needed for this action, when they don't seem to be needed for building the image?

Reply:
This may actually be a holdover from back when the protostellar files were in CNG proper and were part of the `go generate` call.  They could probably be removed now.